docs/src/main/asciidoc/client-customization.adoc

```diff
@@ -0,0 +1,47 @@
+== Client Customization
+
+The library uses the https://github.com/Azure/azure-sdk-for-java[Azure SDK for Java] for connecting to Azure App Configuration and Azure Key Vault. Two interfaces, `ConfiguraitonClientBuilderSetup` and `SecretClientBuilderSetup` are provided to modify the clients. Each interface has a `setup` method that takes in their respective builder along with the `String` value of the URI that the client will connect to. 
+
+[source,java,indent=0]
+----
+public interface ConfigurationClientBuilderSetup {
+    public void setup(ConfigurationClientBuilder builder, String endpoint);
+}
+
+public interface SecretClientBuilderSetup {
+    public void setup(SecretClientBuilder builder, String uri);
+}
+----
+
+This enables modifying the http client and its configurations. This can be used to replace the default `HttpClient`` with one using a proxy for all traffic to App Configuration and Key Vault.
```

Comment:
I would rephrase this to "These interfaces allow for customization of the http client and its configurations, for example, by replacing the default `HttpClient` with another that uses a proxy for all traffic directed to App Configuration and Key Vault. This example can be seen below"